### PR TITLE
feat: framework-agnostic service definitions

### PR DESCRIPTION
**Pull request by aikoven**
This PR adds support for generating rpc-framework-agnostic service definitions.

It allows to build an RPC framework that consumes `ts-proto` generated code, without the need to have framework-specific codegen beforehand.

A framework can still provide type-safe interfaces for the user using TypeScript mapped types.

### CODE REVIEW

File: integration/service-definitions/simple.ts [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+    return message;
+  },
+};
+
+/**
+ * Test
+ *
+ * @deprecated
+ */
+export const TestDefinition = {
```
Comment (aikoven) [inline integration/service-definitions/simple.ts:84]: I've added the `Definition` suffix as per [discussion](https://github.com/stephenh/ts-proto/pull/262#discussion_r606353216).

Though now it seems that the suffix should be `ServiceDefinition`

File: src/options.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@ export type Options = {
   stringEnums: boolean;
   constEnums: boolean;
   outputClientImpl: boolean | 'grpc-web';
-  outputServices: false | 'grpc-js';
+  outputServices: boolean | 'grpc-js';
```
Comment (aikoven) [inline src/options.ts:39]: Not sure whether it should be a separate option. Perhaps someone might want to have both `grpc-js` and generic definitions

Comment (aikoven) [inline src/options.ts:39]: Need to add documentation on the option to the Readme

File: integration/update-bins.sh [hunk 3]
```diff
@@ -4,7 +4,7 @@
 # proto requests into .bin files that then unit tests can pull in directly as needed.
 list=$(find . -name "*.proto" -type f)
 
-for file in $list; do
+for file in service-definitions/*.proto; do
```
Comment (stephenh) [inline integration/update-bins.sh:7]: Looks like this snuck in.

File: src/options.ts (hunk 2 above)
Comment (stephenh) [reply src/options.ts:39]: How about `outputServices: false | 'grpc-js' | 'generic-definitions` and then maybe someday (doesn't have to be now) we could support like csv of `outputServices=grpc-js,generic-definitions`?

**Review by stephenh — APPROVED** [review]

File: integration/service-definitions/simple.ts (hunk 1 above)
Comment (aikoven) [reply integration/service-definitions/simple.ts:67]: Hey @stephenh, thank you for review. What do you think about this one?

File: integration/update-bins.sh (hunk 3 above)
Comment (aikoven) [reply integration/update-bins.sh]: Fixed

File: src/options.ts (hunk 2 above)
Comment (aikoven) [reply src/options.ts]: Done

Comment (aikoven) [reply src/options.ts]: Done

File: integration/service-definitions/simple.ts (hunk 1 above)
Comment (aikoven) [reply integration/service-definitions/simple.ts]: After some thinking, I'm fine with `Definition` suffix as long as we don't overload this term and only use it for services.

**Comment by aikoven** [comment]
Hey @stephenh, any chance you could merge this PR soon? We seem to have settled on the changes. I've already made an integration in `nice-grpc`, and I am ready to publish it as soon as this feature is released.

**Comment by stephenh** [comment]
@aikoven yep, thanks for the ping!

**Comment by aikoven** [comment]
Thanks!

**Pull request merged**